src/components/containers/TraceAccordion.js

```diff
@@ -59,57 +61,47 @@ class TraceAccordion extends Component {
       );
     });
 
-    return Object.keys(fullDataArrayPositionsByTraceType).map((type, index) => {
-      return (
-        <TraceFold
-          key={index}
-          traceIndexes={dataArrayPositionsByTraceType[type]}
-          name={traceTypes(_).find(t => t.value === type).label}
-          fullDataArrayPosition={fullDataArrayPositionsByTraceType[type]}
-        >
-          {this.props.children}
-        </TraceFold>
-      );
-    });
+    return Object.keys(fullDataArrayPositionsByTraceType).map((type, index) => (
+      <TraceFold
+        key={index}
+        traceIndexes={dataArrayPositionsByTraceType[type]}
+        name={traceTypes(_).find(t => t.value === type).label}
+        fullDataArrayPosition={fullDataArrayPositionsByTraceType[type]}
+      >
+        {this.props.children}
+      </TraceFold>
+    ));
   }
 
   renderUngroupedTraceFolds() {
-    if (!this.filteredTraces.length) {
-      return null;
-    }
-
-    return this.filteredTraces.map((d, i) => {
-      return (
+    return (
+      this.filteredTraces.length > 0 &&
+      this.filteredTraces.map((d, i) => (
         <TraceFold
           key={i}
           traceIndexes={[d.index]}
           canDelete={this.props.canAdd}
-          messageIfEmpty={this.props.messageIfEmptyFold}
           fullDataArrayPosition={[this.filteredTracesFullDataPositions[i]]}
         >
           {this.props.children}
         </TraceFold>
-      );
-    });
+      ))
+    );
   }
 
   renderTraceFolds() {
-    if (!this.filteredTraces.length) {
-      return null;
-    }
-
-    return this.filteredTraces.map((d, i) => {
-      return (
+    return (
+      this.filteredTraces.length > 0 &&
+      this.filteredTraces.map((d, i) => (
         <TraceFold
           key={i}
-          traceIndexes={[i]}
+          traceIndexes={[this.filteredTracesFullDataPositions[i]]}
```

Comment:
yeah... im not sure about this, here i think we'd want data indexes not fullData, why do we need that change?

traceIndexes should stay data related, that's what its meaning was throughout the code.

Reply:
i is not index of data, it's index of a filteredTraces array. Perhaps I should make a `filteredTracesDataPositions`?

Reply:
actually, `filteredTracesFullDataPositions` are not relative to fullData, they're relative to base which could be anything